runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -1530,6 +1540,112 @@ J9::CodeGenerator::lowerTreeIfNeeded(
 
    }
 
+void
+J9::CodeGenerator::runGlobalLiveVariablesForGC()
+   {
+   auto comp = self()->comp();
+   auto trHeapMemory = self()->trHeapMemory();
+   auto trMemory = self()->trMemory();
+
+   if (TR::Compiler->om.areValueTypesEnabled() &&
+       (comp->getOption(TR_EnableParanoidOptCheck) || debug("paranoidOptCheck")))
+      comp->verifyCFG(comp->getMethodSymbol());
+
+   if (!comp->getFlowGraph()->getStructure())
+      {
+      comp->getFlowGraph()->setStructure(TR_RegionAnalysis::getRegions(comp));
+      }
+
+   // Because only live locals are mapped for GC, there is normally no need to
+   // make sure locals are cleared to NULL during method prologue.
+   // However, we can have cases where GC-collected locals are live at the start
+   // of the method. These locals will have to be cleared to NULL during method
+   // prologue.
+   // This can happen because of the way we treat non-inlined jsrs.
+   //
+   int32_t numLocals = 0;
+   TR::AutomaticSymbol *p;
+   ListIterator<TR::AutomaticSymbol> locals(&comp->getMethodSymbol()->getAutomaticList());
+   for (p = locals.getFirst(); p != NULL; p = locals.getNext())
+      {
+      // Mark collected locals as initialized. We will reset this property for
+      // any locals that are live at the start of the method.
+      //
+      if (p->isCollectedReference() &&
+          (!comp->getOption(TR_MimicInterpreterFrameShape) ||
+           !comp->areSlotsSharedByRefAndNonRef() ||
+           p->isSlotSharedByRefAndNonRef()))
+         p->setInitializedReference();
+      ++numLocals;
+      }
+
+   if (comp->getOption(TR_EnableAggressiveLiveness))
+      {
+      TR::ParameterSymbol *pp;
+      ListIterator<TR::ParameterSymbol> parms(&comp->getMethodSymbol()->getParameterList());
+      for (pp = parms.getFirst(); pp != NULL; pp = parms.getNext())
+         ++numLocals;
+      }
+
+   // Nothing to do if there are no locals
+   //
+   if (numLocals == 0)
+      return;
+
+   TR_BitVector *liveVars = NULL;
+
+   if ((comp->getOption(TR_EnableOSR) && (comp->getHCRMode() == TR::osr || comp->getOption(TR_FullSpeedDebug)))
+       || !self()->getLiveLocals()) // under OSR existing live locals is likely computed without ignoring OSR uses
```

Comment:
If live locals are already computed before CG by whatever opt, it may also be invalidated by any block split operation. Make sure you invalidate any live local in block split.

Reply:
> Make sure you invalidate any live local in block split.

Does it mean to set CG's livelocals to `NULL` at where the splitPostGRA is used such as `comp->cg()->setLiveLocals(NULL)`?

Reply:
Updated in  29bb422 to invalid the live locals when splitPostGRA is used.

Reply:
We shouldn't trust any livelocals computed earlier given that anything can happen after the computation. But it's impossible to guarantee that anyone changing CFG knows they might break things in CG, and invalidate livelocals. Given that no one sets live locals on CG except GlobalLiveVariableForGC, which is to be moved to CG, I think we should just run the analysis even if CG has live locals.

Reply:
I'll remove the check on `!self()->getLiveLocals() `and the extra invalidation. I'll run some tests first before pushing another commit.

Reply:
I think this whole check should be removed 
```   
if ((comp->getOption(TR_EnableOSR) && (comp->getHCRMode() == TR::osr || comp->getOption(TR_FullSpeedDebug)))
       || !self()->getLiveLocals())
```

Reply:
We still need this condition to adhere to old behavior
```
comp()->getOption(TR_EnableOSR) && (comp()->getHCRMode() == TR::osr || comp()->getOption(TR_FullSpeedDebug))
```

Reply:
> I think we should just run the analysis even if CG has live locals.

With the checking on `getLiveLocals` being removed (~`|| !self()->getLiveLocals()`~ ) since the analysis should be run regardless, wouldn't keeping the rest of the old condition check (`comp()->getOption(TR_EnableOSR) && (...))`)  cause the analysis to run only when OSR is enabled? What if OSR is disabled and getLiveLocals() == NULL or getLiveLocals() returns an already invalid result due to other splitPostGRA before CG?

Reply:
You're right, we should run the analysis regardless given the last condition being always true